app/tests/tests_10/test_external_collection_assets_endpoint.py

```diff
@@ -0,0 +1,373 @@
+        collection = self.collection
+
+        collection_asset = self.factory.create_collection_asset_sample(
+            collection=collection, sample='external-asset', db_create=True
+        )
+
+        response = self.client.get(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset.attr_name]
+            )
+        )
+
+        json_data = response.json()
+
+        self.assertEqual(json_data['href'], f"{TEST_SERVER}/{collection_asset.attr_file}")
+
+    def test_create_collection_asset_with_invalid_external_url(self):
+        collection = self.collection
+
+        collection.allow_external_assets = True
+        collection.save()
+
+        collection_asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': 'this-is-not-a-url'
+        }
+
+        # create the asset
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(400, response)
+        description = response.json()['description']
+        self.assertIn('href', description, msg=f'Unexpected field error {description}')
+
+        self.assertEqual(
+            "Invalid URL provided", description['href'][0], msg="Unexpected error message"
+        )
+
+    def test_create_collection_asset_with_inexistent_external_url(self):
+        collection = self.collection
+
+        collection.allow_external_assets = True
+        collection.external_asset_whitelist = [
+            settings.EXTERNAL_TEST_ASSET_URL, 'https://map.geo.admin.ch'
+        ]
+        collection.save()
+
+        collection_asset_data = {
+            'id': 'clouds.jpg',
+            'description': 'High in the sky',
+            'type': 'image/jpeg',  # specify the file explicitly
+            'href': settings.EXTERNAL_TEST_ASSET_URL
+        }
+
+        # create the asset with an existing one
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        self.assertStatusCode(201, response)
+
+        collection_asset_data['href'] = 'https://map.geo.admin.ch/notexist.jpg'
+        # create the asset with an existing one
+        response = self.client.put(
+            reverse_version(
+                'collection-asset-detail', args=[collection.name, collection_asset_data['id']]
+            ),
+            data=collection_asset_data,
+            content_type="application/json"
+        )
+
+        description = response.json()['description']
```

Comment:
Assert status code

Also at the end of the test check that no asset has been created on the DB